### PR TITLE
fix: properly escape shell special characters in git commits (fixes #2658)

### LINKED ISSUE #2658: Git commit does not properly escape backquote (`)
### What happened?

When Gemini CLI offers to prepare a commit message, it includes markdown including backquotes (`). Those are not properly escaped when the git commit is executed so they end up as command substitutions.

This issue is also a security risk due to unintended command execution.

Example (w/ gemini-2.5-flash):

Image

### What did you expect to happen?

The commit message should be properly escaped, in particular the backquotes (`).

### Client information

<details>

```console
$ gemini /about
╭────────────────────────────────────────╮
│                                        │
│ About Gemini CLI                       │
│                                        │
│ CLI Version  0.1.7                     │
│ Git Commit   55cc809                   │
│ Model        gemini-2.5-flash          │
│ Sandbox      no sandbox                │
│ OS           linux                     │
│ Auth Method  OAuth                     │
│                                        │
╰────────────────────────────────────────╯

```

</details>

### Login information

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by maslinedwin**
## TLDR

This PR fixes a critical security vulnerability where backticks and other shell special characters in git commit messages could lead to command injection attacks. The fix adds proper shell escaping to prevent malicious code execution.

## Dive Deeper

The vulnerability allowed attackers to inject arbitrary commands through git commit messages containing:
- Backticks (`) for command substitution
- `$()` syntax for command substitution  
- Semicolons, pipes, and other shell operators

Our solution implements a comprehensive shell escaping utility that:
1. Wraps all arguments in single quotes to prevent shell interpretation
2. Escapes embedded single quotes using the '\'' pattern
3. Handles edge cases like empty strings and special characters
4. Updates AI prompts to enforce proper escaping

This follows OWASP best practices for preventing command injection vulnerabilities.

## Reviewer Test Plan

1. **Test malicious payloads are properly escaped**:
   ```bash
   # Try committing with dangerous characters
   gemini "Fix the bug in `rm -rf /` handling"
   gemini "Update docs $(echo malicious)"
   gemini "Add feature'; cat /etc/passwd; echo '"
   ```

2. **Verify normal commits still work**:
   ```bash
   gemini "Add new feature for user authentication"
   gemini "Fix: resolve issue with API endpoints"
   ```

3. **Run the test suite**:
   ```bash
   npm test -- packages/core/src/utils/shellEscape.test.ts
   ./integration-tests/git-commit-escaping.test.js
   ```

4. **Check the security documentation**:
   ```bash
   cat docs/security/shell-injection-prevention.md
   ```

## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |
| npx      | ✅  | ❓  | ❓  |
| Docker   | ❓  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

## Linked issues / bugs

- Fixes #2658 - Backticks in git commits messages result in syntax error

### CODE REVIEW

File: packages/core/src/utils/shellEscape.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+/**
+ * Escapes a commit message for safe use with git commit -m
+ * @param message The commit message to escape
+ * @returns The properly escaped commit message
+ */
+export function escapeGitCommitMessage(message: string): string {
+  return escapeShellArg(message);
+}
```
[the remark this answers is not in the document]
Comment (maslinedwin) [reply packages/core/src/utils/shellEscape.ts]: @gemini-code-assist I've addressed the critical Windows security issue:

  - Added cross-platform shell escaping that detects OS and applies appropriate rules
  - POSIX systems: Uses single quotes with '\'' escaping
  - Windows systems: Uses double quotes with ^ escaping for cmd.exe
  - Added comprehensive tests for both platforms
  - Updated security documentation

  The vulnerability is now fixed on all platforms.

[the remark this answers is not in the document]
Comment (maslinedwin) [reply packages/core/src/utils/shellEscape.ts]: /gemini review

**Comment by NTaylorMullen** [comment]
Hey @maslinedwin I think this PR may be out of date now. We recently moved to utilizing a shell escaping package. Feel free to re-open if you think I've missed something!
